resource/docker/network/network_test.go

```diff
@@ -0,0 +1,406 @@
+// Copyright © 2016 Asteris, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package network_test
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/asteris-llc/converge/helpers/comparison"
+	"github.com/asteris-llc/converge/helpers/fakerenderer"
+	"github.com/asteris-llc/converge/helpers/logging"
+	"github.com/asteris-llc/converge/resource"
+	"github.com/asteris-llc/converge/resource/docker/network"
+	dc "github.com/fsouza/go-dockerclient"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/net/context"
+)
+
+// TestNetworkInterface verifies that Network implements the resource.Task
+// interface
+func TestNetworkInterface(t *testing.T) {
+	t.Parallel()
+	defer logging.HideLogs(t)()
+
+	assert.Implements(t, (*resource.Task)(nil), new(network.Network))
+}
+
+// TestNetworkCheck tests the Network.Check function
+func TestNetworkCheck(t *testing.T) {
+	t.Parallel()
+	defer logging.HideLogs(t)()
+
+	nwName := "test-network"
+
+	t.Run("state: absent", func(t *testing.T) {
+		t.Run("network does not exist", func(t *testing.T) {
+			nw := &network.Network{Name: nwName, State: "absent"}
+			c := &mockClient{}
+			nw.SetClient(c)
+			c.On("FindNetwork", nwName).Return(nil, nil)
+
+			status, err := nw.Check(context.Background(), fakerenderer.New())
+			require.NoError(t, err)
+			assert.False(t, status.HasChanges())
+			assert.Equal(t, 1, len(status.Diffs()))
+			comparison.AssertDiff(t, status.Diffs(), nwName, "absent", "absent")
+		})
+
+		t.Run("network exists", func(t *testing.T) {
+			nw := &network.Network{Name: nwName, State: "absent"}
+			c := &mockClient{}
+			nw.SetClient(c)
+			c.On("FindNetwork", nwName).Return(&dc.Network{Name: nwName}, nil)
+
+			status, err := nw.Check(context.Background(), fakerenderer.New())
+			require.NoError(t, err)
+			assert.True(t, status.HasChanges())
+			assert.Equal(t, 1, len(status.Diffs()))
+			comparison.AssertDiff(t, status.Diffs(), nwName, "present", "absent")
```

Comment:
true but it requires a cast (otherwise, I get `cannot use network.StatePresent (type network.State) as type string in argument to comparison.AssertDiff`). makes the lines really long and harder to read (imo). worth it?